docker-compose/dev/mysql8.yaml

```diff
@@ -0,0 +1,105 @@
+version: '3'
+services:
+  mysql:
+    image: mysql:8.0
+    ports:
+      - "3306:3306"
+    environment:
+      - "MYSQL_ROOT_PASSWORD=xdbxdb"
+  
+  # below is for Pulsar        
+  # Start zookeeper
+  zookeeper:
+    image: apachepulsar/pulsar:latest
+    container_name: zookeeper
+    restart: on-failure
+    networks:
+      - pulsar
+    volumes:
+      - ./data/zookeeper:/pulsar/data/zookeeper
+    environment:
+      - metadataStoreUrl=zk:zookeeper:2181
+      - PULSAR_MEM=-Xms256m -Xmx256m -XX:MaxDirectMemorySize=256m
+    command: >
+      bash -c "bin/apply-config-from-env.py conf/zookeeper.conf && \
+             bin/generate-zookeeper-config.sh conf/zookeeper.conf && \
+             exec bin/pulsar zookeeper"
+    healthcheck:
+      test: ["CMD", "bin/pulsar-zookeeper-ruok.sh"]
+      interval: 10s
+      timeout: 5s
+      retries: 30
+
+  # Init cluster metadata
+  pulsar-init:
```

Comment:
Is it possible to put the common steps in the two yamls into a common file?

Reply:
Ahh looks like it’s implemented recently…after .. 9 years https://github.com/compose-spec/compose-spec/pull/363


https://www.docker.com/blog/improve-docker-compose-modularity-with-include/#:~:text=With%20include%20support%2C%20Docker%20Compose,in%20the%20config%20file%20organization.